types/index.d.ts

```diff
@@ -31,52 +33,121 @@ interface AssembleOptions {
   importHook: (filePath: string, type: 'idl', callback: Callback<object>) => void;
 }
 
+export function assemble(args: any): any;
 export function assembleProtocol(filePath: string, opts: Partial<AssembleOptions>, callback: Callback<object>): void;
 export function assembleProtocol(filePath: string, callback: Callback<object>): void;
+export function combine(args: any): any;
 export function createFileDecoder(fileName: string, codecs?: Partial<CodecOptions>): Decoder;
 export function createFileEncoder(filePath: string, schema: any, options?: any): Encoder;
 export function discoverProtocol(transport: Service.Transport, options: any, callback: Callback<any>): void;
 export function discoverProtocol(transport: Service.Transport, callback: Callback<any>): void;
 export function extractFileHeader(filePath: string, options?: any): void;
-export function parse(schemaOrProtocolIdl: string, options?: any): Service.Protocol | Type.Type; // TODO
-export function readProtocol(protocolIdl: string, options?: Partial<ReaderOptions>): Service.Protocol;
+export function infer(args: any): any;
+export function parse(schemaOrProtocolIdl: string, options?: any): Protocol | Type; // TODO
+export function readProtocol(protocolIdl: string, options?: Partial<ReaderOptions>): Protocol;
 export function readSchema(schemaIdl: string, options?: Partial<ReaderOptions>): Schema;
 // TODO streams
+
 // TODO types
+export class Type {
+  constructor(schema: any, opts: any);
+  clone(val: any, opts?: any): any;
+  compare(val1: any, val2: any): number;
+  compareBuffers(buf1: any, buf2: any): number;
+  createResolver(type: any, opts?: any): any;
+  decode(buf: any, pos?: any, resolver?: any): any;
+  encode(val: any, buf: any, pos?: any): any;
+  equals(type: any): any;
+  fingerprint(algorithm?: any): any;
+  fromString(str: any): any;
+  getAliases(): any;
```

Comment:
All the getters will soon be deprecated (and removed in the next major version), in favor of properties:

+ `getAliases()` -> `aliases`
+ `getName(bool)` -> `name` and `branchName`
+ ...

Can you update the declarations accordingly throughout?

Reply:
Aside: https://github.com/mtth/avsc/wiki/API#typetypes-1

I don't see a definition of getBranchType() but an example mentions that method. Should that be `constructor.type`?

Reply:
Good catch, the example should reference `constructor.type` instead. I'll update the documentation.